addons/crm/static/tests/forecast_kanban_tests.js

```diff
@@ -0,0 +1,262 @@
+                            <div><field name="name"/></div>
+                        </t>
+                    </templates>
+                </kanban>`,
+            archs: {
+                "lead,false,search": `
+                    <search>
+                        <filter name="forecast" string="Forecast" context="{'forecast_filter':1}"/>
+                        <filter name='groupby_date' context="{'group_by':'date'}"/>
+                    </search>`
+            },
+            data: {
+                lead: {
+                    fields: {
+                        name: {string: 'Name', type: 'char'},
+                        date: {string: "Date Field", type: 'date'},
+                        datetime: {string: "Datetime Field", type: 'datetime'},
+                    },
+                },
+            },
+            services: { 'fillTemporalService': FillTemporalService },
+            model: 'lead',
+            View: ForecastKanbanView,
+            viewOptions: {
+                context: {
+                    search_default_forecast: true,
+                    search_default_groupby_date: true,
+                    forecast_field: 'date',
+                },
+            },
+            groupBy: ['date'],
+        };
+        this.unPatchDate = await testUtils.mock.patchDate(2021, 9, 1, 0, 0, 0);
+    },
+    afterEach: async function () {
+        await this.unPatchDate();
+    },
+
+}, function () {
+    /**
+     * Since mock_server does not support fill_temporal, 
+     * we only check the domain and the context sent to the read_group, as well
+     * as the end value of the FillTemporal Service after the read_group (which should have been updated in the model)
+     */
+    QUnit.test("Forecast on months, until the end of the year of the latest data", async function (assert) {
+        assert.expect(3);
+
+        this.testKanbanView.data.lead.records = [
+            {id: 1, name: 'Lead 1', date: '2021-01-01'},
+            {id: 2, name: 'Lead 2', date: '2021-02-01'},
+            {id: 3, name: 'Lead 3', date: '2021-11-01'},
+            {id: 4, name: 'Lead 4', date: '2022-01-01'},
+        ];
+        this.testKanbanView.mockRPC = function (route, args) {
+            if (route === '/web/dataset/call_kw/lead/web_read_group') {
+                assert.deepEqual(args.kwargs.context.fill_temporal, {
+                    fill_from: "2021-10-01",
+                    min_groups: 4,
+                });
+                assert.deepEqual(args.kwargs.domain, [
+                    "&", "|",
+                        ["date", "=", false], ["date", ">=", "2021-10-01"],
+                    "|",
+                        ["date", "=", false], ["date", "<", "2023-01-01"],
+                ]);
+            }
+            return this._super.apply(this, arguments);
+        };
+        var kanban = await testUtils.createView(this.testKanbanView);
+        assert.equal(kanban.call('fillTemporalService', 'getFillTemporalPeriod', {
```

Comment:
`assert.equal` can generally be replaced either by `deepEqual` or `strictEqual`

```suggestion
        assert.strictEqual(kanban.call('fillTemporalService', 'getFillTemporalPeriod', {
```